kubernetes-itests/src/test/java/io/fabric8/kubernetes/KubernetesRoleIT.java

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes;
+
+import io.fabric8.kubernetes.api.model.rbac.KubernetesRole;
+import io.fabric8.kubernetes.api.model.rbac.KubernetesRoleList;
+import io.fabric8.kubernetes.api.model.rbac.KubernetesRoleBuilder;
+import io.fabric8.kubernetes.api.model.rbac.KubernetesPolicyRuleBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import org.arquillian.cube.kubernetes.api.Session;
+import org.arquillian.cube.kubernetes.impl.requirement.RequiresKubernetes;
+import org.arquillian.cube.requirement.ArquillianConditionalRunner;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(ArquillianConditionalRunner.class)
+@RequiresKubernetes
+public class KubernetesRoleIT {
+
+  @ArquillianResource
+  KubernetesClient client;
+
+  @ArquillianResource
+  Session session;
+
+  private KubernetesRole kubernetesRole;
+
+  private String currentNamespace;
+
+  @Before
+  public void init() {
+
+    currentNamespace = session.getNamespace();
+
+    KubernetesRole kubernetesRole = new KubernetesRoleBuilder()
+      .withNewMetadata()
+      .withName("job-reader")
+      .endMetadata()
+      .addToRules(0, new KubernetesPolicyRuleBuilder()
+        .addToApiGroups(0,"batch")
+        .addToResourceNames(0,"my-job")
+        .addToResources(0,"jobs")
+        .addToVerbs(0, "get")
+        .addToVerbs(1, "watch")
+        .addToVerbs(2, "list")
+        .build()
+      )
+      .build();
+
+    client.rbac().kubernetesRoles().inNamespace(currentNamespace).createOrReplace(kubernetesRole);
+  }
+
+  @Test
+  public void get() {
+    kubernetesRole = client.rbac().kubernetesRoles().inNamespace(currentNamespace).withName("job-reader").get();
+
+    assertNotNull(kubernetesRole);
```

Comment:
Better to have serializer and deserializer tests than too many assertions.

Reply:
@hrishin Here serializer and deserializer that does not make sense as these are real time tests and when you create real time object, lots of fields get added like creationTimestamp and there are more.

Reply:
@piyush1594 make sense then.